source/Halibut/Transport/Streams/NetworkTimeoutStream.cs

```diff
@@ -30,6 +30,8 @@ public override async ValueTask DisposeAsync()
 
         public override async Task FlushAsync(CancellationToken cancellationToken)
         {
+            ThrowIfAlreadyTimedOut();
```

Comment:
I moved this out from `WrapWithCancellationAndTimeout`, because `WrapWithCancellationAndTimeout` never ended up calling `ThrowIfAlreadyTimedOut`.

This was because `CanTimeout` is called outside `WrapWithCancellationAndTimeout`, which also calls `ThrowIfAlreadyTimedOut`. So when you look at the stack trace, it goes via `CanTimeout`. 

Which is just a little odd. 

Really, this type of thing should be at the public entry point level, not within a private.